property_code_enforcements/make_property_code_enforcements.R

```diff
@@ -26,105 +26,70 @@ raw_data <-
       "Closed - Duplicate Complaint"
     )
   ) |>
-  mutate(SUB_TYPE_DESC = stringr::str_to_lower(SUB_TYPE_DESC),
-         address = stringr::str_to_lower(FULL_ADDRESS), 
-        address = clean_address_text(address)) |> 
+  mutate(description = stringr::str_to_lower(SUB_TYPE_DESC)) |> 
+  select(
+    id = NUMBER_KEY, 
+    date = ENTERED_DATE,
+    description,
+    status = DATA_STATUS_DISPLAY,
+    address = FULL_ADDRESS,
+    lat_jittered = LATITUDE,
+    lon_jittered = LONGITUDE,
+  ) |>
   filter(address != "")
 
-# pull unique addresses to reduce computation time
-d_address <- tibble::tibble(address = unique(raw_data$address))
-
-# convert address character strings into an addr vector
-d_address$addr <- as_addr(d_address$address)
-
-# batch process for memory reasons (because using addr_match_line_one instead of looping over ZIPs)
-i_1 <- seq(1, nrow(d_address), by = 1000)
-i_2 <- i_1 + 999
-
-d_address_list <- purrr::map2(i_1, i_2, ~d_address[.x:.y,])
-d_address_list[[length(i_1)]] <- na.omit(d_address_list[[length(i_1)]])
+# unique addresses
+d_address <- 
+  tibble::tibble(address = unique(raw_data$address)) |>
+  mutate(addr = addr::addr(glue::glue("{address} Anytown XX 00000")))
 
 # match with addr::cagis_addr reference addresses included in the package
-d_address_list <- 
-  purrr::map(
-    d_address_list, 
-    \(x) 
-    x |>
-    mutate(cagis_addr_matches = addr:::addr_match_line_one(addr, cagis_addr$cagis_addr)), 
+d_address_match <- 
+  purrr::map2(
+    seq(from = 1, to = nrow(d_address), by = 100), 
+    c(seq(from = 100, to = nrow(d_address), by = 100), nrow(d_address)),
+    \(x, y) 
+    addr_match_street_name_and_number(
+      x = d_address$addr[x:y], 
+      ref_addr = cagis_addr()$cagis_addr,
+      stringdist_match = "osa_lt_1", 
+      match_street_type = TRUE, 
+      simplify = TRUE
+    ), 
   .progress = TRUE
   )
 
-# collapse back to one tibble
-d <- bind_rows(d_address_list)
+d_address_match_tibble <- purrr::map(d_address_match, \(x) tibble::tibble(cagis_addr = x))
+d_address_match_tibble <- bind_rows(d_address_match_tibble)
 
-d <- 
-  d |>
-  mutate(
-    addr_match_result =
-      case_when(
-        purrr::map_lgl(cagis_addr_matches, is.null) ~ NA,
-        purrr::map_dbl(cagis_addr_matches, vctrs::vec_size) == 0 ~ "no_match",
-        purrr::map_dbl(cagis_addr_matches, vctrs::vec_size) == 1 ~ "single_match",
-        purrr::map_dbl(cagis_addr_matches, vctrs::vec_size) > 1 ~ "multi_match",
-        .default = "foofy"
-      ) |>
-      factor(levels = c("no_match", "single_match", "multi_match"))
-  )
+d_address <- bind_cols(d_address, d_address_match_tibble)
 
-summary(d$addr_match_result) # include in readme (only keep single match)
+saveRDS(d_address, "matched_addr.rds")
```

Comment:
Is this temporary?  Or do you want to track this in git?

Reply:
Yes, this was temporary (to prevent losing the output since the compute time was so long). No longer needed with ZCTA matching